### PR TITLE
[query] fix and enforce that workers transmit stack traces

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

File: hail/src/main/scala/is/hail/backend/spark/SparkBackend.scala [hunk 1]
```diff
@@ -424,7 +425,16 @@ class SparkBackend(
...
           // FIXME: this is broken: the partitionId of SparkTaskContext will be incorrect
-          val result = Try(f(sp.data, SparkTaskContext.get(), theHailClassLoaderForSparkWorkers, fs))
+          val result = try {
+            Success(f(sp.data, SparkTaskContext.get(), theHailClassLoaderForSparkWorkers, fs))
+          } catch {
+            case NonFatal(exc) =>
+              val nowhereStream = new PrintStream(new OutputStream() {
+                def write(b: Int): Unit = {}
+              })
+              exc.printStackTrace(nowhereStream) // Not sure why but this ensures the exception has a stack trace
```
Comment (ehigham) [inline hail/src/main/scala/is/hail/backend/spark/SparkBackend.scala:435]:
```suggestion
              exc.getStackTrace()
              Failure(exc)
```

Comment (ehigham) [reply hail/src/main/scala/is/hail/backend/spark/SparkBackend.scala:435]: ^ This should have the same effect (after reading Throwable.java)

**Review by ehigham — CHANGES_REQUESTED** [review]

**Comment by danking** [comment]
Verifying this works when back ported to 0.2.126

**Comment by danking** [comment]
Confirmed it fixes 0.2.126 as well.

**Review by ehigham — APPROVED** [review]

**Pull request merged**